xdsl/backend/riscv/lowering/convert_memref_to_riscv.py

```diff
@@ -26,13 +28,29 @@
     RewritePattern,
     op_type_rewrite_pattern,
 )
+from xdsl.traits import SymbolTable
 from xdsl.utils.exceptions import DiagnosticException
 
 
 class ConvertMemrefAllocOp(RewritePattern):
+
     @op_type_rewrite_pattern
     def match_and_rewrite(self, op: memref.Alloc, rewriter: PatternRewriter) -> None:
-        raise DiagnosticException("Lowering memref.alloc not implemented yet")
+        assert isinstance(op_memref_type := op.memref.type, memref.MemRefType)
+        size = prod(op_memref_type.get_shape()) * 8
```

Comment:
I think you can't assume that the types size is 8 bytes long, in the example code above you have an i32, which is only 4 bytes long, while an i128 might be more than 8 bytes.

I think supporting this in general will require us to add a way to get the size of a type. I think @Moxinilian worked a bit on this.

Reply:
We have this! I'll add it now

Reply:
Turns out we didn't have this, or I couldn't find it